File: Simple Factory/1. Minimal/Python/main.py
from product_factory import ProductFactory

def main():
    print("Simple Factory in Python")
    print("------------------------")
    print()

    product1 = ProductFactory.createProduct("Product1")
    product2 = ProductFactory.createProduct("Product2")

    product1.operation()
    product2.operation()

if __name__ == "__main__":
    main()

File: Simple Factory/1. Minimal/Java/Main.java
public class Main {
    public static void main(String[] args) {
        System.out.println("Simple Factory in Java");
        System.out.println("----------------------");
        System.out.println();

        Product product1 = ProductFactory.createProduct("Product1");
        Product product2 = ProductFactory.createProduct("Product2");

        product1.operation();
        product2.operation();
    }
}
